packages/opencensus-core/src/stats/recorder.ts

```diff
@@ -14,12 +14,86 @@
  * limitations under the License.
  */
 
-import {AggregationData, Measurement} from './types';
+import {AggregationData, AggregationType, CountData, DistributionData, LastValueData, Measurement, SumData} from './types';
 
 export class Recorder {
   static addMeasurement(
       aggregationData: AggregationData,
       measurement: Measurement): AggregationData {
-    throw new Error('Not Implemented');
+    switch (aggregationData.type) {
+      case AggregationType.DISTRIBUTION:
+        return this.addToDistribution(
+            aggregationData as DistributionData, measurement);
+
+      case AggregationType.SUM:
+        return this.addToSum(aggregationData as SumData, measurement);
+
+      case AggregationType.COUNT:
+        return this.addToCount(aggregationData as CountData, measurement);
+
+      default:
+        return this.addToLastValue(
+            aggregationData as LastValueData, measurement);
+    }
+  }
+
+  private static addToDistribution(
+      distributionData: DistributionData,
+      measurement: Measurement): DistributionData {
+    distributionData.timestamp = Date.now();
+
+    distributionData.count += 1;
+
+    const inletBucket = distributionData.buckets.find((bucket) => {
+      return bucket.lowBoundary <= measurement.value &&
+          measurement.value < bucket.highBoundary;
+    });
+    inletBucket.count += 1;
+
+    if (measurement.value > distributionData.max) {
+      distributionData.max = measurement.value;
+    }
+
+    if (measurement.value < distributionData.min) {
+      distributionData.min = measurement.value;
+    }
+
+    if (distributionData.count === 1) {
+      distributionData.mean = measurement.value;
+    }
+
+    distributionData.sum += measurement.value;
+
+    const oldMean = distributionData.mean;
+    distributionData.mean = distributionData.mean +
```

Comment:
(1) I think this will introduce drift from roundoff errors over time. It might be best to re-compute the mean from `sum / count`. Though I guess there is no way to avoid the same issue for stdev.
(2) Is this formula correct? Do you have a source for this?

Reply:
I like the idea of using just `sum/count`, it's way simpler, but it may introduce overflow problems when `sum` gets too large, really large, which may not be a use case. If this specific situation won't happen, or will be really hard to happen, I'm okay with `sum/count`.

This formula comes from manipulating the mean definition itself, there is a nice explanation [here](http://datagenetics.com/blog/november22017/index.html). This is also [how Go does](https://github.com/census-instrumentation/opencensus-go/blob/master/stats/view/aggregation_data.go#L138). But, again, I'm on board with `sum/count` if that problem won't happen.

Reply:
Overflow is a good concern that I didn't think about. In that case I am more ambivalent, and incrementing might just as good given that it stays consistent with stdev. It will not be completely accurate for Double values but maybe that is more acceptable.